server/src/main/java/org/apache/uniffle/server/buffer/ShuffleBufferManager.java

```diff
@@ -431,20 +436,28 @@ private Map<String, Set<Integer>> pickFlushedShuffle() {
     long expectedFlushSize = highWaterMark - lowWaterMark;
     long pickedFlushSize = 0L;
     int printIndex = 0;
+    int printIgnoreIndex = 0;
     int printMax = 10;
     for (Map.Entry<String, AtomicLong> entry : sizeList) {
       long size = entry.getValue().get();
-      pickedFlushSize += size;
       String appIdShuffleIdKey = entry.getKey();
-      addPickedShuffle(appIdShuffleIdKey, pickedShuffle);
-      // print detail picked info
-      if (printIndex < printMax) {
-        LOG.info("Pick application_shuffleId[{}] with {} bytes", appIdShuffleIdKey, size);
-        printIndex++;
-      }
-      if (pickedFlushSize > expectedFlushSize) {
-        LOG.info("Finish flush pick with {} bytes", pickedFlushSize);
-        break;
+      if (size > this.shuffleFlushThreshold) {
+        pickedFlushSize += size;
```

Comment:
What will it happen if shuffle server only have small ios?

Reply:
If the shuffle server reached the high watermark, but there only are small I/Os, what will it happen?

Reply:
Is the behavior different from origin one?

Reply:
I mean that you change the flush strategy instead of adding an extra configuration. Could we keep the origin behavior by the configuration?